### PR TITLE
Fix filter logic

### PR DESCRIPTION
**Pull request by kjhenner**

### CODE REVIEW

File: lib/survey_gizmo/api/response.rb [hunk 1]
```diff
@@ -42,7 +42,9 @@ def parsed_answers
         # Strip out "Other" answers that don't actually have the "other" text (they come back as two responses - one
         # for the "Other" option_id, and then a whole separate response for the text given as an "Other" response.
         if k =~ /\[question\((\d+)\),\s*option\((\d+)\)\]/
-          !answers.keys.any? { |key| key =~ /\[question\((#{$1})\),\s*option\("(#{$2})-other"\)\]/ }
+          if answers.keys.any? { |key| key =~ /\[question\((#{$1})\),\s*option\("(#{$2})-other"\)\]/ }
+            next false
+          end
```
Comment (apurvis) [inline lib/survey_gizmo/api/response.rb]: wouldn't you need
```
else 
  true
```

to get the `select` to work properly?  also not sure why you would need a `next`...

**Comment by kjhenner** [comment]
Ugh, looks like it's deleting too many option answers now...

**Comment by apurvis** [comment]
yeah the `select` block isn't getting a `true` by default...

File: lib/survey_gizmo/api/response.rb (hunk 1 above)
Comment (kjhenner) [inline lib/survey_gizmo/api/response.rb]: Yeah, that makes sense, but isn't working for me.

Comment (apurvis) [inline lib/survey_gizmo/api/response.rb]: best thing to do is probably craft a spec out of your issue

**Comment by apurvis** [comment]
specs are failing as well...

this project is in dire need of webmock/VCR specs though; maybe this is the time to try to add them.

p.s. i'm also concerned that SG might have different response formats for different question types.  based on nothing but suspicion really but they have done weirder things.

**Comment by apurvis** [comment]
this looks correct but what exactly was the issue?

**Comment by kjhenner** [comment]
It seems like those regex capture groups were leaking out of their proper scope and causing some kind of funkiness. Not sure I could say exactly what was happening, but explicitly naming them resolves the issues I was seeing.

**Comment by apurvis** [comment]
hmm i wouldn't be totally shocked

**Comment by kjhenner** [comment]
So the `.any?` is iterating over all the keys and matching against them, so I think it's going to have those `$1` and `$2` assigned as the question_id and option_id for the first thing it looks at, but as soon as there's a successful match against something, those `$1` and `$2` variables will be reassigned as the capture groups there.

**Comment by apurvis** [comment]
it's also maybe possible that if you are using some scope like `parsed_answers.each { do_stuff }` that somehow the `do_stuff` is mucking up `$1` and `$2`?  is that relevant to how you are testing?

either way if this fixes your issue and the specs pass, 👍

**Comment by apurvis** [comment]
also i think i actually get what you are saying now... brutal regex problems, inc.

**Comment by apurvis** [comment]
can you bump the minor version number on this PR?

**Comment by apurvis** [comment]
this also might be the culprit if its not an issue with the `.any?` block
https://github.com/kjhenner/survey-gizmo-ruby/blob/f78e9766c0cfd03187a738dbf64cd5b19c127cb9/lib/survey_gizmo/api/answer.rb#L21

**Comment by apurvis** [comment]
@jarthod actually this is the PR we should release after, not the other one.

**Comment by jarthod** [comment]
Nice catch, these capture groups were definitely being overwritten by the second regexp interpolating them, using named capture is a nice fix and reads better. You could also have simply removed the capture groups in the second regexp as they are not used. Released `6.2.2`

**Comment by apurvis** [comment]
in other news, i found out that SG sometimes leaves out the option altogether when giving two answers.

```
<SurveyGizmo::API::Answer:0x007f0449fb1c48
 @answer_text="Other - Please explain",
 @key="[question(5)]",
 @question_id=5,
 @response_id=172,
 @submitted_at=Wed, 29 Jun 2016 17:58:30 EDT -04:00,
 @survey_id=xxx,
 @value="Other - Please explain">
#<SurveyGizmo::API::Answer:0x007f0449fb0c58
 @answer_text="I understood... ",
 @key="[question(5), option(\"10017-other\")]",
 @option_id=10017,
 @other_text="I understood...  ",
 @question_id=5,
 @response_id=172,
 @submitted_at=Wed, 29 Jun 2016 17:58:30 EDT -04:00,
 @survey_id=xxx,
 @value="I understood...  ">
```

note there are two answers.  one has the key `"[question(5)]"` and the other has the key `"[question(5), option(\"10017-other\")]"`.  not sure what the hell to do about that ugh.

**Pull request merged**